libs/ledgerjs/packages/context-module/src/forward-domain/domain/ForwardDomainContextLoader.test.ts

```diff
@@ -0,0 +1,83 @@
+import { LoaderOptions } from "../../shared/model/LoaderOptions";
+import { Transaction } from "../../shared/model/Transaction";
+import { ForwardDomainContextLoader } from "./ForwardDomainContextLoader";
+
+describe("ForwardDomainContextLoader", () => {
+  const transaction = {} as Transaction;
+
+  beforeEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  describe("load function", () => {
+    it("should return an empty array when no domain or registry", async () => {
+      const options = {} as LoaderOptions;
+
+      const loader = new ForwardDomainContextLoader({ getDomainNamePayload: jest.fn() });
+      const promise = () => loader.load(transaction, options);
+
+      expect(promise()).resolves.toEqual([]);
+    });
+
+    it("should throw an error when no registry", async () => {
+      const options = { options: { forwardDomain: { domain: "test.eth" } } } as LoaderOptions;
+
+      const loader = new ForwardDomainContextLoader({ getDomainNamePayload: jest.fn() });
+      const promise = () => loader.load(transaction, options);
+
+      expect(promise()).rejects.toThrow(
+        new Error(
+          "[ContextModule] ForwardDomainLoader: Invalid combination of domain and registry. Either both domain and registry should exist",
+        ),
+      );
+    });
+
+    it("should throw an error when no domain", async () => {
+      const options = { options: { forwardDomain: { registry: "ens" } } } as LoaderOptions;
+
+      const loader = new ForwardDomainContextLoader({ getDomainNamePayload: jest.fn() });
+      const promise = () => loader.load(transaction, options);
+
+      expect(promise()).rejects.toThrow(
+        new Error(
+          "[ContextModule] ForwardDomainLoader: Invalid combination of domain and registry. Either both domain and registry should exist",
+        ),
+      );
+    });
+
+    // TODO: complete all the edge case here
```

Comment:
I would probably just add 1 test for domain > max length as it feels like it's the only real other business rule we have ?

Reply:
Added 👍